respa_admin/views/resources.py

```diff
@@ -424,21 +422,33 @@ def forms_valid(
         resource_image_formset,
         resource_accessibility_formset,
     ):
-        self.object = form.save()
-        self._save_resource_purposes()
-        self._delete_extra_images(resource_image_formset)
-        self._save_resource_images(resource_image_formset)
-        self.save_period_formset(period_formset_with_days)
-        self._save_or_update_opening_hours_via_period_templates()
-        self._delete_extra_resource_accessibility(resource_accessibility_formset)
-        self._save_resource_accessibility(resource_accessibility_formset)
+        try:
+            with transaction.atomic():
+                self.object = form.save()
```

Comment:
`ValidationError` may be raised if the formset is misconfigured (e.g. `TOTAL_FORMS` out of sync due to JS removing a form). We don't want to return a 500 error to the user, instead wrap in transaction to ensure no invalid data is saved and re-display the form.